cmd/snap-confine/mount-support.c

```diff
@@ -707,9 +707,7 @@ void sc_ensure_shared_snap_mount(void)
 {
 	if (!is_mounted_with_shared_option("/")
 	    && !is_mounted_with_shared_option(SNAP_MOUNT_DIR)) {
-		sc_do_mount(SNAP_MOUNT_DIR, SNAP_MOUNT_DIR, "none",
-			    MS_BIND | MS_REC, 0);
-		sc_do_mount("none", SNAP_MOUNT_DIR, NULL, MS_SHARED | MS_REC,
-			    NULL);
+		die("cannot proceed, %s is not mounted with shared option",
```

Comment:
Is there a risk to break existing distros here that may not set / shared?

Reply:
I mainly did it to open my eyes in case something was severely broken. Since it didn't fail on 14.04 I think we're good. The mainly have snaps fail and closed later than late and open.

Reply:
I'm still wondering the same as Michael: do we have a good basis to believe that this won't break people in the wild? We support quite a few environments right now.

Also, the "cannot proceed," prefix may be dropped (it's implicit in almost every error).